src/Optimization/hiopIterate.cpp

```diff
@@ -65,6 +65,12 @@ hiopIterate::hiopIterate(const hiopNlpFormulation* nlp_)
   sxu = x->alloc_clone();
   sdl = d->alloc_clone();
   sdu = d->alloc_clone();
+  sx_arg1_ = x->alloc_clone();
```

Comment:
I would allocate on demand, in the `adjust_small_slacks` method. I think bound pushing occurs rather infrequently ...